### DISCUSSION TITLE
Custom Headers

### DISCUSSION THREAD

**Discussion by masitings** [dedoc/scramble#299] [Q&A] [answered]
Hi there,

I want to ask, is there a way for me to put custom header in security in AppServiceProvider ? Currently, im using 2 layer of Auth, First one is using Sanctum and second was using App auth. So, i want to put the custom headers somewhere near the Token authenticated. Is it possible ? 

![image](https://github.com/dedoc/scramble/assets/29356430/c9e21498-5007-488d-b524-5ce86788a20c)

**Comment by mojtabasoori** [comment, marked as answer]
hi,

i add custom header with extension.

```php
use Dedoc\Scramble\Support\RouteInfo;
use Dedoc\Scramble\Support\Generator\Schema;
use Dedoc\Scramble\Support\Generator\Operation;
use Dedoc\Scramble\Support\Generator\Parameter;
use Dedoc\Scramble\Extensions\OperationExtension;
use Dedoc\Scramble\Support\Generator\Types\StringType;

class AddCustomHeader  extends OperationExtension
{
    public function handle(Operation $operation, RouteInfo $routeInfo)
    {
        $operation->addParameters([
            Parameter::make('X-custom-id', 'header')
                ->setSchema(
                    Schema::fromType(new StringType())
                )
                ->required(true)
                ->example("925fdfe2-745c-45d4-aaaf-40f3455c0510")
        ]);
    }
}

```

**Comment by masitings** [comment]
@mojtabasoori Its worked! Thanks. For those of you who want to add the same thing, you need to add this as extensions in scramble.php config files.
![image](https://github.com/dedoc/scramble/assets/29356430/b3e3a46e-7861-4444-86a4-440fe7650e31)

**Reply by hondamitrajayarayi** [reply]
I've done the same thing, but why is the parameter input in the documentation not showing up?
is there something missing?
![image](https://github.com/user-attachments/assets/c45eef6c-fa8e-4319-b18d-c05d7dadc142)

**Reply by romalytvynenko** [reply]
@hondamitrajayarayi I have no idea. Share enough info so I can reproduce.

**Reply by hondamitrajayarayi** [reply]
I added an extension to config/scramble.php
`
'extensions' => [
        AddCustomHeader::class,
    ],
`

then, in the AddCustomHeader.php file, I equate the contents with the one above.
Did I miss anything?

**Reply by romalytvynenko** [reply]
@hondamitrajayarayi no, it should've worked. I have no idea why it doesn't. Feel free to create a repro repository so I can help. Otherwise I cannot

**Reply by romalytvynenko** [reply]
@hondamitrajayarayi try it without call to `example` method. If it doesn't help, I'd need repro repository.

**Reply by hondamitrajayarayi** [reply]
I've tried without the `example `method but it still doesn't work.

here's the repository
https://github.com/hondamitrajayarayi/scrumble-api-docs

**Reply by romalytvynenko** [reply]
@hondamitrajayarayi notice you place the extension class in **Exceptions** folder, but you import the file from **Extensions** folder...

**Reply by hondamitrajayarayi** [reply]
@romalytvynenko  thank you very much, I was not careful enough
